### ISSUE TITLE
DeclarationOrderCheck false positive

### ISSUE THREAD

**Issue by boris-petrov** [checkstyle/checkstyle#5155]
```java
/* default */ final FirebirdSql firebirdSql;
```

This gives a DeclarationOrderCheck warning which is not right. This too:

```java
/* default */ static final EventHandler handler = (event, subject, arguments) -> {
```

**Comment by rnveach** [comment]
@boris-petrov Please see http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug on how bug reports should be given.
We need to see a full example and your configuration.

**Comment by boris-petrov** [comment]
@rnveach - here's an example for the first case:

```java
public class Test {
	private final String foo = "";
	/* default */ final String bar = "";
}
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="DeclarationOrder"/>
    </module>
</module>
```

**Comment by romani** [comment]
@boris-petrov , please follow template - https://github.com/checkstyle/checkstyle/blob/master/.github/ISSUE_TEMPLATE.md , all points are critical.
We close issues that are unclear.

**Comment by boris-petrov** [comment]
/var/tmp $ javac YOUR_FILE.java

No output

/var/tmp $ cat YOUR_FILE.java

```java
public class Test {
	private final String foo = "";
	/* default */ final String bar = "";
}
```

/var/tmp $ cat config.xml

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="DeclarationOrder"/>
    </module>
</module>
```

/var/tmp $ java -Duser.language=en -Duser.country=US -jar checkstyle-X.XX-all.jar -c config.xml YOUR_FILE.java

```
%  java -Duser.language=en -Duser.country=US -jar checkstyle-8.2-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /home/boris/Test.java:3:23: Variable access definition in wrong order. [DeclarationOrder]
Audit done.
Checkstyle ends with 1 errors.
```

I'm not sure what was unclear about my previous post and why I had to do the same steps and show you the same output.

**Comment by romani** [comment]
and now all become clear .... violation is correct

http://checkstyle.sourceforge.net/config_coding.html#DeclarationOrder
>  First the public class variables, then protected, then package level (no access modifier), and then private. 

in your test, there is `private` first and then `default`.
Expected is `... package level (no access modifier), and then private.`
